packages/modules/imodel-browser/src/containers/iModelGrid/IModelGrid.tsx

```diff
@@ -116,43 +119,55 @@ export const IModelGrid = ({
   const tileApiOverrides = apiOverrides
     ? { serverEnvironmentPrefix: apiOverrides.serverEnvironmentPrefix }
     : undefined;
-  return iModels.length === 0 && noResultsText ? (
-    <NoResults text={noResultsText} />
-  ) : (
-    <GridStructure>
-      {fetchStatus === DataStatus.Fetching ? (
-        <>
-          <IModelGhostTile />
-          <IModelGhostTile />
-          <IModelGhostTile />
-        </>
-      ) : (
-        <>
-          {iModels?.map((iModel) => (
-            <IModelHookedTile
-              key={iModel.id}
-              iModel={iModel}
-              iModelOptions={iModelActions}
-              accessToken={accessToken}
-              onThumbnailClick={onThumbnailClick}
-              apiOverrides={tileApiOverrides}
-              useTileState={useIndividualState}
-              {...tileOverrides}
-            />
-          ))}
-          {fetchMore ? (
-            <>
-              <InView onChange={fetchMore}>
+
+  const renderIModelGridStructure = () => {
+    return (
+      <GridStructure>
+        {fetchStatus === DataStatus.Fetching ? (
+          <>
+            <IModelGhostTile />
+            <IModelGhostTile />
+            <IModelGhostTile />
+          </>
+        ) : (
+          <>
+            {iModels?.map((iModel) => (
+              <IModelHookedTile
+                key={iModel.id}
+                iModel={iModel}
+                iModelOptions={iModelActions}
+                accessToken={accessToken}
+                onThumbnailClick={onThumbnailClick}
+                apiOverrides={tileApiOverrides}
+                useTileState={useIndividualState}
+                {...tileOverrides}
+              />
+            ))}
+            {fetchMore ? (
+              <>
+                <InView onChange={fetchMore}>
+                  <IModelGhostTile />
+                </InView>
                 <IModelGhostTile />
-              </InView>
-              <IModelGhostTile />
-              <IModelGhostTile />
-            </>
-          ) : null}
-        </>
-      )}
-    </GridStructure>
-  );
+                <IModelGhostTile />
+              </>
+            ) : null}
+          </>
+        )}
+      </GridStructure>
+    );
+  };
+
+  const renderComponent = () => {
+    if (iModels.length === 0 && noResultsText && emptyStateComponent) {
+      return emptyStateComponent;
+    }
```

Comment:
The empty component will not show why there is no results, in most case, if you have no results because of an error, you should show it.

Reply:
I am not able to understand here, emptyStateComponent will be shown when there is no data and also in case of error. I can see in case of error we are setting imodels to an empty array
![image](https://user-images.githubusercontent.com/113327714/207803332-fe54350e-2ee5-47ff-941b-c0ed0398f7e0.png)

Reply:
Yes, and the status will be "FetchFailed", in which case, the default component will display "An error occured", so the user know that we didnt find anything because of an error, not because there is no iModel in the iTwin: 
![image](https://user-images.githubusercontent.com/1904889/207887183-39ed793c-f385-4302-9390-a0435c945282.png)

The way that we display the provided component now does not let this component know about this fact and will override the error message, in the storybook example, "There are no iModels to show." will be displayed, regardless of the fact that there was an error, leading the user to believe that they lost all their work...
![image](https://user-images.githubusercontent.com/1904889/207887514-9dad11b6-8580-45c4-95e4-0344bdaf2a57.png)

Reply:
Now showing emptyStateComponent only when there are no iModels

Reply:
I think it's likely that people using this feature will want to control the error page as well (so the `emptyStateComponent` need to receive the information), but we can introduce a context and hook then if this is ever requested without breaking the API. Let's keep it simple like this for now.